listenbrainz/db/dump.py

```diff
@@ -301,6 +349,96 @@ def create_public_dump(location, dump_time, threads=DUMP_DEFAULT_THREAD_COUNT):
     )
 
 
+def create_feedback_dump(location, dump_time, threads=DUMP_DEFAULT_THREAD_COUNT):
+    """ Create a spark format dump of user listen and user recommendation feedback.
+    """
+    return _create_dump(
+        location=location,
+        dump_type='feedback',
+        tables=[],
+        dump_time=dump_time,
+        threads=threads,
+    )
+
+
+def dump_user_feedback(connection, location):
+    """ Carry out the actual dumping of user listen and user recommendation feedback.
+    """
+
+    with connection.begin() as transaction:
+
+        # First dump the user feedback
+        result = connection.execute(sqlalchemy.text("""
+            SELECT musicbrainz_id, recording_msid, score, r.created,
+                   EXTRACT(YEAR FROM r.created) AS year,
+                   EXTRACT(MONTH FROM r.created) AS month,
+                   EXTRACT(DAY FROM r.created) AS day
+              FROM recording_feedback r
+              JOIN "user"
+                ON r.user_id = "user".id
+          ORDER BY created"""))
+
+        last_day = ()
+        todays_items = []
+
+        while True:
+            row = result.fetchone()
+            today = (row[4], row[5], row[6]) if row else ()
+            if (not row or today != last_day) and len(todays_items) > 0:
+                full_path = os.path.join(location, "feedback", "listens", "%02d" % int(last_day[0]),
+                                         "%02d" % int(last_day[1]), "%02d" % int(last_day[2]))
+                os.makedirs(full_path)
+                with open(os.path.join(full_path, "data.json"), "wb") as f:
```

Comment:
small thing - apparently this concept of writing multiple lines of json to a single file has started to be called "jsonl": https://jsonlines.org/, perhaps for clarity we could name this file `data.jsonl` instead

Reply:
then I think we should update all of our dumps in on go, not just one being in consistent from another. I would say we do this when we get rid of MSIDs from spark, which will basically be a complete refactor of the CF spark code. That would be a good time to introduce this change.